test/uk/gov/hmrc/helptosavestub/controllers/SquidControllerSpec.scala

```diff
@@ -0,0 +1,983 @@
+
+  private val goodCreateAccountMap: Map[String, String] =
+    Map("forename" -> "Donald",
+      "surname" -> "Duck",
+      "address1" -> "1",
+      "address2" -> "Test Street 2",
+      "address3" -> "Test Place 3",
+      "address4" -> "Test Place 4",
+      "address5" -> "Test Place 5",
+      "postcode" -> "GIR 0AA",
+      "countryCode" -> "GB",
+      "NINO" -> "WM123456C",
+      "birthDate" -> "19920509",
+      "communicationPreference" -> "02",
+      "phoneNumber" -> "+44111 111 111",
+      "emailAddress" -> "dduck@email.com",
+      "registrationChannel" -> "online")
+
+  private def generateJson(variants: Seq[(String, String)] = Seq()): JsValue = {
+    Json.toJson(Map("createAccount" -> (goodCreateAccountMap ++ variants)))
+  }
+
+  private def generateJsonFromMap(m: Map[String, String]): JsValue = {
+    Json.toJson(Map("createAccount" -> m))
+  }
+
+  private def fakeRequest = FakeRequest().withJsonBody(generateJson()).withHeaders((CONTENT_TYPE, "application/json"))
+
+  private def makeFakeRequest(json: JsValue) = FakeRequest().withJsonBody(json).withHeaders((CONTENT_TYPE, "application/json"))
+
+  private def buildRequest(json: JsValue) = FakeRequest("", "", FakeHeaders(), json.toString())
+
+  "Squid Controller" must {
+
+    "return 415 if the mime type is not application/json" in {
+      val fakeRequestWithoutJson = FakeRequest("POST", "/help-to-save-stub/create-account")
+      val result = new SquidController(messagesApi).createAccount()(fakeRequestWithoutJson)
+      status(result) shouldBe 415
+    }
+
+    "return 200 when requested" in {
+      val result = new SquidController(messagesApi).createAccount()(fakeRequest)
+      status(result) shouldBe 200
+    }
+
+    "The request should return some content" in {
+      //Note that if this is returning a blank page, the controller configuration in application.conf is broken
+      val result = new SquidController(messagesApi).createAccount()(fakeRequest)
+      status(result)
+      val len = result.body.contentLength.getOrElse(0L)
+      len shouldBe 0
+    }
+
+    "If the stub is sent a request with no JSON content it should return a 400" in {
+      val fakeRequestWithoutJson = FakeRequest("POST", "/help-to-save-stub/create-account").withHeaders((CONTENT_TYPE, "application/json"))
+      val result = new SquidController(messagesApi).createAccount()(fakeRequestWithoutJson)
+      status(result) shouldBe 400
+    }
+
+    "If the stub is sent a request with no JSON content it should have an Error object in the response with the errorMessageId set as AAAA0002" in {
+      val fakeRequestWithoutJson = FakeRequest("POST", "/help-to-save-stub/create-account").withHeaders((CONTENT_TYPE, "application/json"))
+      val result: Future[Result] = new SquidController(messagesApi).createAccount()(fakeRequestWithoutJson)
+      status(result)
+      val json: JsValue = contentAsJson(result)
+      val errorMessageId = (json \ "error" \ "errorMessageId").get.asOpt[String]
```

Comment:
you could do something like `json.validate[ErrorClass]` and use the resulting `JsResult`